### DISCUSSION TITLE
Youtube Video Always Downloads in German

### DISCUSSION THREAD

**Discussion by Toastinator** [aclap-dev/video-downloadhelper#389] [Help and Q&A] [answered]
### Your question

I'm using Firefox, Downloading a YouTube Video that Play's in English. The Video is fine, But it always downloads the German Audio for some Reason. I want to downloaded at the Highest Quality Possible. Any help would be awesome! Thanks!

MURDER DRONES - Episode 1: PILOT
[https://www.youtube.com/watch?v=mImFz8mkaHo&t=581s](URL)

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): b197dc8833d544ed
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\~~~~~~",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by Mattmjs** [comment, marked as answer]
Go in settings>more>video qualities, untick Ignore low quality videos. Then look for an extra download down the bottom.

**Reply by Toastinator** [reply]
That worked! But the Video only has Audio for the First Half. The Second half is Silent. Any Thoughts? Thanks for your help so far!

**Comment by GrampaWildWilly** [comment]
This is not really a solution but I did manage to get the complete 2560x1440 webm of this.  It is rather a tortuous process & I'm going to leave out a few details.  The complete details are in a thread with the title "How to download from YouTube" back on the other forum on Google.

For some reason, your link above is not a link to YouTube.  Click it.  It comes back to a GitHub page that says the page doesn't exist.  I had to copy the URL off this page & paste it into the URL bar of a new Firefox window.  I don't know what's wrong with it.

I also downloaded the video with VDH & could get only the German sound track.  There's a ton of languages to choose from when you're playing the video back in the YouTube web page.  I tried a few of them & they did switch the audio I was hearing from one language to another.

Well, it's time for Plan B.

Once I got onto the YouTube page, I opened the Network Monitor.  The preferred object to look for is an entry of type json with player in the file name.  There was no such object.  So I did the usual evasive maneuver.  I entered the name of the video into the YouTube search field at the top of the page & executed the search.  That gave me a search results page that included this video as one of the search results.  Then being careful to hover the mouse over just that search result did give me a player json.  Hover.  Not click.  I just wanted to get the player json to appear in the Network Monitor.  Once I had that, I dragged the top edge of the Network Monitor up to the top of the page, covering the search results, making them not visible.  That's because the rest of this exercise is carried out strictly in the Network Monitor.

It turns out there's something like 70 or 80 adaptive formats of this thing.  There's a bunch of video/webm variants, a bunch of video/mp4 variants, & as you might expect, even more audio/webm & audio/mp4 variants than videos.  The first object in this bunch was a video/webm of resolution 2560x1440.  This appeared the be the highest resolution variant available in either webm or mp4.  So I extracted the URL for that video track from the Network Monitor:

```
https://rr2---sn-vgqsrnzd.googlevideo.com/videoplayback?expire=1712754610&ei=UjsWZp6wGe-U2_gPhKabmAI&ip=2603%3A6011%3Ac306%3Aa341%3A85f3%3Aeda2%3A9901%3A4d43&id=o-AB4PoSB4fcSaln6aYJeg3qFETaPFseNwrFdibJ6eQ8mG&itag=271&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C271%2C278%2C394%2C395%2C396%2C397%2C398%2C399%2C400&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=Xc&mm=31%2C29&mn=sn-vgqsrnzd%2Csn-vgqsknz6&ms=au%2Crdu&mv=m&mvi=2&pl=33&initcwndbps=1276250&bui=AaUN6a1dUne5k-iwoe_ipJEsShKz4Uds3_06d3_yLW1rQnjpecJEJROtH3CJmy-S1E1EMMLFxVShYma0&spc=UWF9f1B_aELBotobUME00wIqVswgQIOLsehHAnCxDfOt_FRl9udLp207DFLZ&vprv=1&svpuc=1&mime=video%2Fwebm&ns=iD8W-WrUrD4KAcmitZBhw10Q&gir=yes&clen=746939153&dur=1615.267&lmt=1707702486025912&mt=1712732683&fvip=5&keepalive=yes&c=WEB&sefc=1&txp=4532434&n=15Ef16NiUXTitEA-&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAPWmVLR9DnOld6yqAQ0cThx5_L5uq5MT0uytybBm_JFBAiEA0fxEwR4RxWqvIU-75w_3dSJ4kuksKeebxgeDRkyHbVk%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=ALClDIEwRgIhAMc1sgU7CwTjNHdfBcR69p0teQIZzZ8NF635q0kfF4A0AiEAn3nIL5yv68GFlnYC44SiVBi7kOE-ZCGMh7ubc8izIio%3D
```

That is unlikely to work for anybody else since it probably has identifiers in it that connect it to my visit to YouTube.  Plus it almost certainly expires after I don't know how long.  In any case, I opened a separate Firefox tab on that URL.  After some hesitation, Firefox rendered that URL with its media player.  There was no audio.  This is as expected.  YouTube supplies video & audio separately.  The Firefox media player indicated it was an object of the correct duration & showed the first frame of the video.  But the audio control was a tiny speaker with a line through it, meaning there was no audio with this.  As it happened, VDH offered to download this.  So I launched that.  It is still running.  The VDH status menu is telling me it's going to take another 10 hours to complete.  Resource Monitor is telling me it's progressing at about 43-44 thousand bytes per second.  Not meg.  Not gig.  Thousand.  I'll probably cancel it shortly.

Then I started trawling through audio/webm variants in the Network Monitor.  I wanted the video/webm so I was going to get the audio/webm.  When I expanded any audio/webm variant, among the mostly unintelligible information it displayed in the Response section of the Network Monitor was an indication of the language of that audio track.  I had to plow through variants for Japanese, Chinese, Russian, Filipino, Korean, Thai, French, Indonesian, I don't remember how many, before I finally stumbled upon one that said it was English.  I actually found 3 variants that claimed to be English.

```
https://rr2---sn-vgqsrnzd.googlevideo.com/videoplayback?expire=1712754610&ei=UjsWZp6wGe-U2_gPhKabmAI&ip=2603%3A6011%3Ac306%3Aa341%3A85f3%3Aeda2%3A9901%3A4d43&id=o-AB4PoSB4fcSaln6aYJeg3qFETaPFseNwrFdibJ6eQ8mG&itag=249&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=Xc&mm=31%2C29&mn=sn-vgqsrnzd%2Csn-vgqsknz6&ms=au%2Crdu&mv=m&mvi=2&pl=33&initcwndbps=1276250&bui=AaUN6a1dUne5k-iwoe_ipJEsShKz4Uds3_06d3_yLW1rQnjpecJEJROtH3CJmy-S1E1EMMLFxVShYma0&spc=UWF9f1B_aELBotobUME00wIqVswgQIOLsehHAnCxDfOt_FRl9udLp207DFLZ&vprv=1&svpuc=1&xtags=acont%3Doriginal%3Alang%3Den&mime=audio%2Fwebm&ns=iD8W-WrUrD4KAcmitZBhw10Q&gir=yes&clen=10210711&dur=1615.281&lmt=1707699905969373&mt=1712732683&fvip=5&keepalive=yes&c=WEB&sefc=1&txp=4532434&n=15Ef16NiUXTitEA-&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cxtags%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIgUPLEkBQSDxj-kgNZNZl99HTw6ra6ox9r_IQY31DSV8MCIQCSdlrqxltuJhGKYL-TKnblnury0a4MZ0DJDDYLKlnwGA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=ALClDIEwRgIhAMc1sgU7CwTjNHdfBcR69p0teQIZzZ8NF635q0kfF4A0AiEAn3nIL5yv68GFlnYC44SiVBi7kOE-ZCGMh7ubc8izIio%3D

https://rr2---sn-vgqsrnzd.googlevideo.com/videoplayback?expire=1712754610&ei=UjsWZp6wGe-U2_gPhKabmAI&ip=2603%3A6011%3Ac306%3Aa341%3A85f3%3Aeda2%3A9901%3A4d43&id=o-AB4PoSB4fcSaln6aYJeg3qFETaPFseNwrFdibJ6eQ8mG&itag=250&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=Xc&mm=31%2C29&mn=sn-vgqsrnzd%2Csn-vgqsknz6&ms=au%2Crdu&mv=m&mvi=2&pl=33&initcwndbps=1276250&bui=AaUN6a1dUne5k-iwoe_ipJEsShKz4Uds3_06d3_yLW1rQnjpecJEJROtH3CJmy-S1E1EMMLFxVShYma0&spc=UWF9f1B_aELBotobUME00wIqVswgQIOLsehHAnCxDfOt_FRl9udLp207DFLZ&vprv=1&svpuc=1&xtags=acont%3Doriginal%3Alang%3Den&mime=audio%2Fwebm&ns=iD8W-WrUrD4KAcmitZBhw10Q&gir=yes&clen=13395266&dur=1615.281&lmt=1707699905971660&mt=1712732683&fvip=5&keepalive=yes&c=WEB&sefc=1&txp=4532434&n=15Ef16NiUXTitEA-&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cxtags%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhAIkO-iFkOrlCPJ5C034Wv3WziA8Y-KC3dq7IruqPLxvfAiEA0BuXvqxVp6wq7eWWlQmiemCX8kz7xhngKgQqtESO34w%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=ALClDIEwRgIhAMc1sgU7CwTjNHdfBcR69p0teQIZzZ8NF635q0kfF4A0AiEAn3nIL5yv68GFlnYC44SiVBi7kOE-ZCGMh7ubc8izIio%3D

https://rr2---sn-vgqsrnzd.googlevideo.com/videoplayback?expire=1712754610&ei=UjsWZp6wGe-U2_gPhKabmAI&ip=2603%3A6011%3Ac306%3Aa341%3A85f3%3Aeda2%3A9901%3A4d43&id=o-AB4PoSB4fcSaln6aYJeg3qFETaPFseNwrFdibJ6eQ8mG&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=Xc&mm=31%2C29&mn=sn-vgqsrnzd%2Csn-vgqsknz6&ms=au%2Crdu&mv=m&mvi=2&pl=33&initcwndbps=1276250&bui=AaUN6a1dUne5k-iwoe_ipJEsShKz4Uds3_06d3_yLW1rQnjpecJEJROtH3CJmy-S1E1EMMLFxVShYma0&spc=UWF9f1B_aELBotobUME00wIqVswgQIOLsehHAnCxDfOt_FRl9udLp207DFLZ&vprv=1&svpuc=1&xtags=acont%3Doriginal%3Alang%3Den&mime=audio%2Fwebm&ns=iD8W-WrUrD4KAcmitZBhw10Q&gir=yes&clen=26325167&dur=1615.281&lmt=1707699905972704&mt=1712732683&fvip=5&keepalive=yes&c=WEB&sefc=1&txp=4532434&n=15Ef16NiUXTitEA-&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cxtags%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRgIhALkEgd8a7m5G7W_V9_FkNpxROwtzA-SHLqBw7-Ios6GYAiEA3-kucPXGB1WYhsPkaidD6YqYRX4SDfNyxL3N0k5x0hQ%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=ALClDIEwRgIhAMc1sgU7CwTjNHdfBcR69p0teQIZzZ8NF635q0kfF4A0AiEAn3nIL5yv68GFlnYC44SiVBi7kOE-ZCGMh7ubc8izIio%3D
```

Each one was a different "quality."  YouTube labels these things as low, medium, & high quality.  I imagine this is their way of attaching a convenient label to audio of varying bit rates.  There were 3 variants of varying qualities for each language.  It was a long & tedious search to find the 3 English audio tracks.  I opened each of these 3 audio URLs in its own tab & used VDH to download them.  Each download was as slow as that video I mentioned above, but audio tracks being what they are, they did all complete within a few minutes.  I didn't actually listen to any of them because I was going to try something different.

I went back to the video tab again & instead of using VDH to download the video, I popped up the context menu on the Firefox media player & executed Save Video As.  This got Firefox to start downloading the video in parallel with the VDH download I've still got running.  It was about the same speed of download.

Then I went back to the audio tab for the third audio URL I've quoted above.  Once again, I popped up the context menu on the Firefox media player & executed Save Audio As.  Launching this download seemed to goose the download of the video track & they both completed in very short order.

I merged the video/webm track with the audio/webm track.  This played fine in VLC, with both video & audio all the way to the end, although it being something I'm utterly not interested in, I just sampled it at intervals, both to make sure it was playing audio & video, & to make sure the language of the audio track was English.

This is way too ridiculously complicated for any but the most dedicated & determined user to bother doing.  I offer it more as a proof of concept.  It shows that this multilingual video with its unusually large array of languages can be downloaded with any of the languages on offer.  I did it with English but it can be done the same way I did it with any of the other languages.  Can VDH do under program control what I did manually in the Network Monitor?  I can't say.  I don't work for VDH.  I'm just another user.  I would hope the answer is yes, if not today, some day.  But I can't actually give that answer.

**Comment by thescrapyarduk** [comment]
Still downloading audio in German, can download the video perfectly using Internet Download Manager I use as my bulk file downloader but that also struggles with some videos that Video Download Helper catches fine

This link IS in English, YouTube again, but downloads in German audio only

https://www.youtube.com/watch?v=uHeWKPYWATY


It may not be a priority to the developers, but those who pay for the extension and get the incorrect audio might consider it a priority


Maybe a simpler workaround for the extension, but in the meantime I use my other downloader for problem videos from YouTube. Obviously YouTube is messing things up again, maybe on purpose

**Reply by jcfrog** [reply]
It is a priority for developers, it's fixed but we had issues with publication. Sorry, still needs a few days.

**Reply by thescrapyarduk** [reply]
Ok, Thanks for the update

I'm sure it will be appreciated by those with incorrect audio downloaded

It might even help others with the audio issues
